### PR TITLE
New property for products - Closes #67

### LINKED ISSUE #67: Add 'new' property to products

### PR DESCRIPTION
**Pull request by peter-gerhard**
@schleichardt @lauraluiz
I add the new property and also an extra sale property (since the template used if priceOld isPresent before and it shouldnt know it)

### CODE REVIEW

File: product-catalog/app/productcatalog/services/CategoryServiceImpl.java [hunk 1]
```diff
@@ -46,4 +47,13 @@ public CategoryServiceImpl(final CategoryTree categories) {
     private Optional<Category> expandCategory(final Reference<Category> categoryRef) {
         return categories.findById(categoryRef.getId());
     }
+
+    public boolean categoryIsInNew(Reference<Category> category) {
+        final Reference<Category> categoryNew = this.categories.findByExternalId("1").get().toReference();
+        return categoryIsAncestorOf(categoryNew, category);
```
Comment (peter-gerhard) [inline product-catalog/app/productcatalog/services/CategoryServiceImpl.java]: I don't like the way the new category is obtained but its just dirty. If you have a better idea for now, pls tell me.

**Comment by peter-gerhard** [comment]
@schleichardt @lauraluiz pls review the last commit

**Comment by lauraluiz** [comment]
Besides the comment, :+1:

**Comment by peter-gerhard** [comment]
@lauraluiz  done

File: product-catalog/app/productcatalog/pages/ProductDataFactory.java [hunk 2]
```diff
@@ -130,13 +126,19 @@ private DetailData localizedStringsToDetailData(final LocalizedString localizedS
         return new DetailData(label, "");
     }
 
-    private Optional<Price> getPriceOld(final Optional<Price> priceOpt) {
-        return priceOpt.flatMap(price -> Optional.ofNullable(price.getDiscounted()).map(discountedPrice -> price));
+    private Optional<Price> getOldPrice(final Price price) {
```
Comment (schleichardt) [inline product-catalog/app/productcatalog/pages/ProductDataFactory.java]: Not necessary to change, but the `javax.annotation.Nonnull` would help the caller to identify, that null is passable.

I think it is a good change to remove optional from parameters.

**Comment by schleichardt** [comment]
Ah `New property for products ` should be `"New" property for products `

File: product-catalog/app/productcatalog/pages/ProductDataFactory.java (hunk 2 above)
Comment (schleichardt) [inline product-catalog/app/productcatalog/pages/ProductDataFactory.java]: sorry, I meant Nullable

**Pull request merged**